src/app/qgsfieldcalculator.cpp

```diff
@@ -482,3 +483,8 @@ void QgsFieldCalculator::setPrecisionMinMax()
   mOutputFieldPrecisionSpinBox->setMinimum( minPrecType );
   mOutputFieldPrecisionSpinBox->setMaximum( qMax( minPrecType, qMin( maxPrecType, mOutputFieldWidthSpinBox->value() ) ) );
 }
+
+void QgsFieldCalculator::helpRequest()
+{
+  QgsHelp::openHelp( "working_with_vector/attribute_table.html#field-calculator" );
```

Comment:
This is indeed the direct link to the field calculator section but I'd keep using working_with_vector/attribute_table.html#editing-attribute-values because:
* it's one level up,
* has an introduction which explains the other ways to edit attribute data (which do not have a help button associated); it could show user that there are better and quicker way to edit data than the field calculator (calculation bar?)
* and field calculator section is the first one so we are not far from the expected section